realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java

```diff
@@ -79,47 +81,63 @@ public AuthenticateResponse refreshUser(Token userToken, URI serverUrl, URL auth
     @Override
     public LogoutResponse logout(Token userToken, URL authenticationUrl) {
         try {
-            String requestBody = LogoutRequest.revoke(userToken).toJson();
-            return logout(buildLogoutUrl(authenticationUrl), requestBody);
+            String requestBody = LogoutRequest.create(userToken).toJson();
+            return logout(buildActionUrl(authenticationUrl, ACTION_LOGOUT), requestBody);
         } catch (Exception e) {
             return LogoutResponse.from(new ObjectServerError(ErrorCode.UNKNOWN, e));
         }
     }
 
-    private static URL buildLogoutUrl(URL authenticationUrl) {
+    @Override
+    public ChangePasswordResponse changePassword(Token userToken, String newPassword, URL authenticationUrl) {
+        try {
+            String requestBody = ChangePasswordRequest.create(userToken, newPassword).toJson();
+            return changePassword(buildActionUrl(authenticationUrl, ACTION_CHANGE_PASSWORD), requestBody);
+        } catch (Exception e) {
```

Comment:
The `Error` will be silently suppressed by the executor service as well i think, catch `Throwable` instead? Or actually maybe it is better that we override `ThreadPoolExecutor.afterExecute` to just throw all uncaught throwables and terminated the process -- when those happen, it cannot be recovered anymore. Then it could be anther PR.